### PR TITLE
Feature main

### PR DESCRIPTION
**Pull request by DersuG**
Create the `main.py` launch point of the program.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import dearpygui.dearpygui as gui
+import serial
+from data_controllers.iliad_data_controller import IliadDataController
+
+class App:
+    def __init__(self) -> None:
+        # Start UI:
+        gui.create_context()
+        gui.create_viewport(title='Iliad Ground Control', width=600, height=300)
```
Comment (PeterHall16) [inline app.py:9]: Nit: I think we're going with "OSUGC" as the name?? Not certain

File: app.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            gui.set_primary_window(MAIN_WINDOW, True)
+
+        gui.show_viewport()
+        gui.maximize_viewport()
+
+        # Start serial ports and stuff:
+        self.iliad = IliadDataController()
+        self.config = {
+            'port_name': 'COM2',
+            'port_baud_rate': 9600,
```
Comment (PeterHall16) [inline app.py:22]: Is this baud rate accurate?

File: data_controllers/data_controller.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        This is usually raised when DataController.open() fails.
+        """
+        pass
+
+    # Returns a dictionary with different config options.
+    def get_config(self) -> dict[str]:
+        return {}
+    
+    # Sets config options from the dictionary passed in.
```
Comment (PeterHall16) [inline data_controllers/data_controller.py:22]: What format would these be stored in? Like could we just have a bunch of txt files or something to pass in for different configurations we might want?

File: data_controllers/iliad_data_controller.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+from utils import packet_util
+import crc
+
+class IliadDataController(serial_data_controller.SerialDataController):
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        self.data_buffer = bytearray()
+        self.checksum_calculator = crc.CrcCalculator(crc.Crc16.CCITT)
```
Comment (PeterHall16) [inline data_controllers/iliad_data_controller.py:13]: Have we verified this is a good checksum algorithm?

File: data_controllers/iliad_data_controller.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    def update(self) -> None:
+        if self.is_open():
+            
+            # Poll serial port and put anything there into data_buffer
+            if self.port.in_waiting > 0:
+                data = self.port.read_all()
+                self.data_buffer += bytearray(data)
+            
+            # If the buffer has enough data, try to parse some of it
+            if len(self.data_buffer) > 10:
```
Comment (PeterHall16) [inline data_controllers/iliad_data_controller.py:24]: Where did 10 come from? Is that arbitrary?

File: data_controllers/iliad_data_controller.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                packet_checksum: int
+
+                # Parse header:
+                if len(self.data_buffer) - self.idx_cursor >= 2:
+                    packet_types_bytes = self.data_buffer[self.idx_cursor:(self.idx_cursor + 2)]
+                    (packet_types_raw,) = struct.unpack('>h', packet_types_bytes)
+                    packet_types: list[int] = list(packet_util.get_packet_types(packet_types_raw))
+                    self.idx_cursor += 2
+                
+                for packet_type in [ # TODO: Turn constants into an enum.
```
Comment (PeterHall16) [inline data_controllers/iliad_data_controller.py:42]: Need to add other types eventually (not urgent)

File: monitor.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+        elif is_front(packet_util.PACKET_TYPE_COORDINATES, packet_types) and len(stream) > 8:
+            packet_types.pop(0)
+            data_bytes = stream[:8]
+            stream = stream[8:]
+            packet[packet_util.PACKET_TYPE_COORDINATES] = struct.unpack('>ff', data_bytes)
+
+            checksum_target += bytearray(data_bytes)
+        
+        elif is_front(packet_util.PACKET_TYPE_C, packet_types) and len(stream) > 4:
```
Comment (PeterHall16) [inline monitor.py:107]: What are types C and D?

**Review by PeterHall16 — COMMENTED** [review]
Left some comments. Overall looks good and ready to merge once we get a second reviewer.

File: app.py (hunk 2 above)
Comment (DersuG) [reply app.py:22]: No, just a placeholder for testing.

File: data_controllers/data_controller.py (hunk 3 above)
Comment (DersuG) [reply data_controllers/data_controller.py:22]: This is sort of a temporary thing until we get a menu/options gui going, but rn it expects `dictionary[str, Any]`. We'll probably use something like toml files for storing options and app-specific data.

Comment (PeterHall16) [reply data_controllers/data_controller.py:22]: Sounds good

File: data_controllers/iliad_data_controller.py (hunk 4 above)
Comment (DersuG) [reply data_controllers/iliad_data_controller.py:13]: Kind of. CRC-16 is enough for our max possible packet size. CCITT is the recommended default by that package.

File: data_controllers/iliad_data_controller.py (hunk 5 above)
Comment (DersuG) [reply data_controllers/iliad_data_controller.py:24]: 10 was our min packet size. We'll need to update this to accommodate for timestamps. It isn't super important though, pretty sure everything will work with other values, although if it's too big we might have some latency.

File: monitor.py (hunk 7 above)
Comment (DersuG) [reply monitor.py:107]: Placeholders for testing.

File: data_controllers/iliad_data_controller.py (hunk 5 above)
Comment (PeterHall16) [reply data_controllers/iliad_data_controller.py:24]: Min packet size is a good way to do it.

**Review by PeterHall16 — APPROVED** [review]
All good

**Pull request merged**